File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices)
    {
        int low =INT_MAX;
        
        int lux=0;
        int n = prices.size();
        for(int i =0 ; i<n ;i++)
        {
            low = min(low,prices[i]);
            lux = max(prices[i]-low,lux);
        }
        return lux;
    }
    
    
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        left=0
        right=1
        maxprofit=0
        while right<len(prices):
            curprofit = prices[right]-prices[left]
            if prices[left]<prices[right]:
                maxprofit=max(maxprofit, curprofit)

            else:
                left =right
            
            right+=1
        
        return maxprofit
